lib/codecs/src/encoding/format/logfmt.rs

```diff
@@ -0,0 +1,77 @@
+use bytes::BytesMut;
+use serde::{Deserialize, Serialize};
+use tokio_util::codec::Encoder;
+use vector_common::encode_logfmt;
+use vector_core::{event::Event, schema};
+
+/// Config used to build a `LogfmtSerializer`.
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct LogfmtSerializerConfig;
+
+impl LogfmtSerializerConfig {
+    /// Creates a new `LogfmtSerializerConfig`.
+    pub const fn new() -> Self {
+        Self
+    }
+
+    /// Build the `LogfmtSerializer` from this configuration.
+    pub const fn build(&self) -> LogfmtSerializer {
+        LogfmtSerializer
+    }
+
+    /// The schema required by the serializer.
+    pub fn schema_requirement(&self) -> schema::Requirement {
+        // While technically we support `Value` variants that can't be losslessly serialized to
+        // logfmt, we don't want to enforce that limitation to users yet.
+        schema::Requirement::empty()
+    }
+}
+
+/// Serializer that converts an `Event` to bytes using the logfmt format.
+#[derive(Debug, Clone)]
+pub struct LogfmtSerializer;
+
+impl LogfmtSerializer {
+    /// Creates a new `LogfmtSerializer`.
+    pub const fn new() -> Self {
+        Self
+    }
+}
+
+impl Encoder<Event> for LogfmtSerializer {
+    type Error = vector_core::Error;
+
+    fn encode(&mut self, event: Event, buffer: &mut BytesMut) -> Result<(), Self::Error> {
+        match event {
+            Event::Log(_) => {
+                let string = encode_logfmt::to_string(event.as_log().as_map())?;
+                buffer.extend_from_slice(string.as_bytes());
+            }
+            Event::Metric(_) => (),
```

Comment:
Could we open up a follow up issue to track adding trace support to all of the relevant encoders?

Reply:
I think with #12561 this will not be an issue an more. In this case we would return `DataType::Log` for this codec

Reply:
Ah, yes 👍